### PR TITLE
Feat/flying cat in gameplay

### PR DESCRIPTION
**Pull request by GitGitarra**

### CODE REVIEW

File: VatSpace/cats.lua [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+end
+
+local function createCatInSpace()
+        cat = display.newImage("resources/graphics/cat.png")
+        cat.speed = 0
+        cat.hasTime = true
+        cat.isVisible = false
+        randomizeFieldsFor(cat)
+        cat.x = math.random(display.contentWidth + 50, display.contentWidth + 350)
```
Comment (s-kania) [inline VatSpace/cats.lua]: 2x same operations while creating cats

File: VatSpace/cats.lua [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        M.cat = cat
+end
+
+function M.move() 
+        a = M.cat
+        moveCat(a)
+        if a.x < -50 then  
+            M.destroy()
+            createCatInSpace()
+            randomizeFieldsFor(a)
```
Comment (s-kania) [inline VatSpace/cats.lua]: so if you destroying cats every time maybe its not necessary to have randomizeFields function?

File: VatSpace/ui.lua [hunk 3]
```diff
@@ -229,26 +251,67 @@ function M.showHelp()
...
+    --     x = display.contentCenterX,
+    --     y = display.contentCenterY + 5,
+    --     width = 180,
+    --     font = "munro_small.ttf",   
+    --     fontSize = 15,
+    --     align = "center"
+    -- }
+    -- local creditsText = display.newText( options )
+    -- creditsText:setFillColor(0.156, 0.133, 0.156)
+    -- credits_page_group:insert(creditsText)
```
Comment (s-kania) [inline VatSpace/ui.lua]: if its not for usage please remove this commented code 😄

File: VatSpace/cats.lua [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+        end
+    end
+    return true
+end
+
+local function randomizeFieldsFor(cat)
+    cat.isVisible = false
+    visibilty = {false, false, true}
+    x = math.random(1, 3)
+    cat.isVisible = visibilty[x]
```
Comment (s-kania) [inline VatSpace/cats.lua]: its not necessary to create table with false,false,true etc. what if we want for example 0.03 chance for something?
probably the best solution is use ternary operator:
cat.isVisible = math.random(1,3) > 2 and true or false
and maybe that is too big chance for cat? but I didn't see this feature in action so I leave it to you 😸

File: VatSpace/cats.lua [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+            showExtraTime(cat.x + 10, cat.y - 25)
+            gv.time = gv.time + 10
+            cat.hasTime = false
+        end
+    end
+    return true
+end
+
+local function randomizeFieldsFor(cat)
+    cat.isVisible = false
```
Comment (s-kania) [inline VatSpace/cats.lua]: not necessary line

**Review by s-kania — CHANGES_REQUESTED** [review]
I left some comments, please take look at it 😸

File: VatSpace/cats.lua (hunk 4 above)
Comment (GitGitarra) [reply VatSpace/cats.lua]: Thanks! That's great idea (however of course I would'n use a table to create 0.03 chance for something :P )  I did few manual test and it's like with "random", you never know, sometimes there is 3 cats in row and sometimes 30 sec without cat. I recommend to leave it like this for now.

File: VatSpace/cats.lua (hunk 2 above)
Comment (GitGitarra) [reply VatSpace/cats.lua]: 👍 changed

File: VatSpace/ui.lua (hunk 3 above)
Comment (GitGitarra) [reply VatSpace/ui.lua]: It's second version of a credits text, if we not gonna use it for sure then I can remove it.

**Review by s-kania — APPROVED** [review]

**Pull request merged**